src/acceleration/BaseQNAcceleration.cpp

```diff
@@ -298,16 +310,15 @@ void BaseQNAcceleration::performAcceleration(
 
   if (_firstIteration && (_firstTimeWindow || _forceInitialRelaxation)) {
     PRECICE_DEBUG("   Performing underrelaxation");
-    _oldXTilde    = _values;    // Store x tilde
-    _oldResiduals = _residuals; // Store current residual
+    _oldPrimaryXTilde    = _primaryValues;    // Store x tilde of primary data
+    _oldXTilde           = _values;           // Store x tilde of primary and secondary data
+    _oldPrimaryResiduals = _primaryResiduals; // Store current residual of primary data
 
     // Perform constant relaxation
     // with residual: x_new = x_old + omega * res
     _residuals *= _initialRelaxation;
-    _residuals += _oldValues;
-    _values = _residuals;
+    _values = _oldValues + _residuals;
```

Comment:
Is this correct? Now `_residuals` has a different value.

Reply:
From the test results I didn't see difference. If you mean that the `_residuals` is not updated here, I see that the `_residuals` would be calculated from `_values` and `_oldValues` by updating differences and is not stored like `_values`, so the step here still works.